actions/publish-github-pages/action.yml

```diff
@@ -1,10 +1,106 @@
 name: 'Publish GitHub Pages'
-description: 'Publish the Hugo-based CloudPosse docs website to GitHub Pages.'
+description: 'Publish the Hugo-based Cloud Posse docs website to GitHub Pages.'
+inputs:
+  content:
+    description: 'comma-separated list of directories in the top level of the repo that contains documentation'
+    required: false
+  css:
+    description: 'custom CSS to insert into custom.scss'
+    required: false
+  debug:
+    description: 'whether to turn on debug printouts/functionality'
+    required: false
+  hugo_url:
+    description: 'URL of the Hugo site after deployment'
+    required: false
+  hugo_publish_dir:
+    description: 'directory in the repo that GitHub Pages will deploy to'
+    required: false
+  hugo_config:
+    description: 'location of to-be-written Hugo config file (actual location not important)'
+    required: false
+  htmltest_config:
+    description: 'location of to-be-written htmltest config file (actual location not important)'
+    required: false
+  github_token:
+    description: 'github token'
+    required: false
+  github_actor:
+    description: 'github actor'
+    required: false
+  github_pages_hugo_path:
+    description: 'location where the repo containing the Hugo files will be checked out to'
+    required: false
+  github_pages_pull_branch:
+    description: 'the branch of the repo which contains the documentation'
+    required: false
+  staging_dir:
+    description: 'directory to collate all docs and Hugo files in'
+    required: false
 runs:
-#  using: 'docker'
-#  image: 'Dockerfile'
   using: 'composite'
   steps:
-    - run: ${{ github.action_path }}/entrypoint.sh
+    - id: checkout-repos
+      run: |
+           pip install GitPython pyyaml
+           python -c 'from checkout_and_collate import *; checkout_repos()'
       shell: bash
-#    - run: ./actions/publish-github-pages/entrypoint.sh
+      env:
+        PYTHONPATH: ${{ github.action_path }}
+        GITHUB_PAGES_PUSH_PATH: ${{ github.action_path }}/${{ env.GITHUB_PAGES_DIRECTORY }}
+        INPUT_GITHUB_PAGES_PULL_BRANCH: ${{ inputs.GITHUB_PAGES_PULL_BRANCH }}
+        INPUT_GITHUB_PAGES_HUGO_PATH: ${{ inputs.GITHUB_PAGES_HUGO_PATH }}
+        INPUT_DEBUG: ${{ inputs.DEBUG }}
+        INPUT_STAGING_DIR: ${{ inputs.STAGING_DIR }}
+    - id: insert-custom-css
+      run: |
+           export CSS=${CSS:-$INPUT_CSS}
+           export GITHUB_PAGES_HUGO_PATH=${GITHUB_PAGES_HUGO_PATH:-$INPUT_GITHUB_PAGES_HUGO_PATH}
+           echo -e "$CSS" >> ${GITHUB_PAGES_HUGO_PATH}/themes/cloudposse/src/scss/custom.scss
+      shell: bash
+      env:
+        INPUT_CSS: ${{ inputs.CSS }}
+        INPUT_GITHUB_PAGES_HUGO_PATH: ${{ inputs.GITHUB_PAGES_HUGO_PATH }}
+    - id: collate-files
+      run: |
+           pip install GitPython pyyaml
+           python -c 'from checkout_and_collate import *; collate_files()'
+      shell: bash
```

Comment:
Instead, set the default shell, and use `-e` so it exits on error. See other workflows for example.

Reply:
There's no top-level `shell:` option in custom github actions, but it looks like `shell: bash` means they use `bash --noprofile --norc -eo pipefail {0}` for the step. (https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell)

Reply:
You mean we cannot use:
```yaml
defaults:
  run:
    shell: bash +e -x -o pipefail {0}
```